slack/slack.go

```diff
@@ -52,15 +52,21 @@ func PingSlackChannel(ctx *ankh.ExecutionContext) error {
 		Username: username,
 	}
 
-	channelId, err := getSlackChannelIDByName(api, ctx.SlackChannel)
-	if err != nil {
-		return err
+	channels := make(map[string]string, len(ctx.AnkhConfig.Slack.Channels))
+	for _, ch := range ctx.AnkhConfig.Slack.Channels {
+		channelId, err := getSlackChannelIDByName(api, ch)
+		if err != nil {
+			return err
+		}
+		channels[ch] = channelId
 	}
 
-	if !ctx.DryRun {
-		_, _, err = api.PostMessage(channelId, slack.MsgOptionAttachments(attachment), slack.MsgOptionPostMessageParameters(messageParams))
-	} else {
-		ctx.Logger.Infof("--dry-run set so not sending message '%v' to slack channel %v", messageText, ctx.SlackChannel)
+	for name, id := range channels {
+		if !ctx.DryRun {
+			_, _, err = api.PostMessage(id, slack.MsgOptionAttachments(attachment), slack.MsgOptionPostMessageParameters(messageParams))
```

Comment:
This error is not checked inside the loop so only the last channel error will be returned/checked from this function.

Reply:
I'm not sure about go conventions for handling multiple possible errors and returning, but this will collect all the messages and return a single error.

Reply:
Or this: https://github.com/hashicorp/go-multierror